enhancements/local-storage/subprovisioner-integration-into-lvms.md

```diff
@@ -0,0 +1,255 @@
+- Implementing a new CSI driver from scratch.
+- Integrating the Subprovisioner CSI driver into TopoLVM.
+
+### Risks and Mitigations
+- There is a risk of increased maintenance burden by integrating a new CSI driver into LVMS without gaining traction
+  - tested separately in the Subprovisioner project as pure CSI Driver similar to TopoLVM and within LVMS with help of QE
+    - we will not GA the solution until we have a clear understanding of the maintenance burden. The solution will stay in TechPreview until then.
+- There is a risk that Subprovisioner is so different from TopoLVM that behavior changes can not be accomodated in the current CRD
+  - we will scrap this effort for integration and look for alternative solutions if the integration is not possible with reasonable effort.
+- There is a risk that Subprovisioner is gonna break easily as its a really young project
+  - we will not GA the solution until we have a clear understanding of the stability of the Subprovisioner project. The solution will stay in TechPreview until then.
+
+## Proposal
+
+The proposal is to extend the LVMCluster CRD with a new deviceClass policy field that can be used to provision shared storage via Subprovisioner.
+We will use this field as a hook in lvm-operator, our orchestrating operator, to provision shared storage via Subprovisioner instead of TopoLVM.
+Whenever LVMCluster discovers a new deviceClass with the Subprovisioner associated policy, it will create a new CSI driver deployment for Subprovisioner and configure it to use the shared storage deviceClass.
+As such, it will handover the provisioning of shared storage to the Subprovisioner CSI driver. Also internal engineering such as sanlock orchestration will be managed by the driver.
+
+
+### Workflow of Subprovisioner instantiation via LVMCluster
+
+1. The user is informed of the intended use case of Subprovisioner, and decides to use it for its multi-node capabilities before provisioning Storage
+2. The user configures LVMCluster with non-default values for the Volume Group and the deviceClass policy field
+3. The lvm-operator detects the new deviceClass policy field and creates a new CSI driver deployment for Subprovisioner.
+4. The Subprovisioner CSI driver is configured to use the shared storage deviceClass, initializes the global lock space, and starts provisioning shared storage.
+5. The user can now provision shared storage via Subprovisioner on the OpenShift Container Platform cluster.
+6. The user can also provision regular TopoLVM deviceClasses side-by-side with shared storage deviceClasses in the same cluster. Then, TopoLVM gets provisioned side-by-side.
+
+## Design Details for `LVMCluster CR extension`
+
+API scheme for `LVMCluster` CR:
+
+```go
+
+  // The DeviceAccessPolicy type defines the accessibility of the create lvm2 volume group backing the deviceClass. 
+  type DeviceAccessPolicy string
+  
+  const (
+    DeviceAccessPolicyShared DeviceAccessPolicy = "shared"
```

Comment:
It's not clear to me which lines are being added/modified. Further below there are lines that start with an extra "+" to indicate they are being added. Could you check that "+" is consistently applied everywhere so it's easy to understand what is being changed?

Reply:
It is not, I already noticed some lines where its not applied. Thanks for pointing out, need to go over this again.

Reply:
I went over this again and checked the lines for changes. should now contain everything